docs/ucxx/Makefile

```
@@ -0,0 +1,19 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
```

```suggestion
SPHINXOPTS    = --fail-on-warning
```

At https://github.com/rapidsai/ucxx/actions/runs/17733583363/job/50390582115?pr=510, it looks like these are building without warnings.

I think we should add `--fail-on-warning`  to ensure it stays that way. That'll turn warnings into loud errors, so they'll be noticed in CI (https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-W).